src/index.js

```diff
@@ -37,42 +43,107 @@ export default function performTextSubstitution(element, preferenceOverrides = n
   if (!process || !process.type === 'renderer') throw new Error(`Not in an Electron renderer context`);
   if (process.platform !== 'darwin') throw new Error(`Only supported on OS X`);
 
-  systemPreferences = systemPreferences || remote.systemPreferences;
+  ipcRenderer = ipcRenderer || electron.ipcRenderer;
+  systemPreferences = systemPreferences || electron.remote.systemPreferences;
 
   if (!systemPreferences || !systemPreferences.getUserDefault) {
     throw new Error(`Electron ${process.versions.electron} is not supported`);
   }
 
-  let textPreferences = preferenceOverrides || readSystemTextPreferences();
-  let replacementItems = getReplacementItems(textPreferences);
+  ipcRenderer.send(registerForPreferenceChangedIpcMessage);
 
-  let inputEvent = new SerialDisposable();
-  inputEvent.setDisposable(addInputListener(element, replacementItems));
+  window.addEventListener('beforeunload', () => {
+    d(`Window unloading, unregister any listeners`);
+    ipcRenderer.send(unregisterForPreferenceChangedIpcMessage);
+  });
 
-  let subscriptionIds = [];
+  let currentAttach = assignDisposableToListener(element, preferenceOverrides);
 
-  for (let preferenceChangedKey of textPreferenceChangedKeys) {
-    subscriptionIds.push(systemPreferences.subscribeNotification(preferenceChangedKey, () => {
-      d(`Got a ${preferenceChangedKey}`);
-      let newTextPreferences = preferenceOverrides || readSystemTextPreferences();
+  ipcRenderer.on(preferenceChangedIpcMessage, () => {
+    d(`User modified text preferences, reattaching listener`);
+    assignDisposableToListener(element, preferenceOverrides, currentAttach);
+  });
 
-      if (didTextPreferencesChange(textPreferences, newTextPreferences)) {
-        d(`User modified text preferences, reattaching listener`);
+  return currentAttach;
+}
 
-        let newReplacementItems = getReplacementItems(newTextPreferences);
-        inputEvent.setDisposable(addInputListener(element, newReplacementItems));
-      }
-    }));
-  }
+/**
+ * Subscribes to text preference changed notifications and notifies listeners
+ * in renderer processes. This method must be called from the main process, and
+ * should be called before any renderer process calls `performTextSubstitution`.
+ *
+ * @return {Disposable}  A `Disposable` that will clean up everything this method did
+ */
+export function listenForPreferenceChanges() {
+  if (!process || !process.type === 'browser') throw new Error(`Not in an Electron browser context`);
+  if (process.platform !== 'darwin') throw new Error(`Only supported on OS X`);
+
+  ipcMain = ipcMain || electron.ipcMain;
+  systemPreferences = systemPreferences || electron.systemPreferences;
```

Comment:
❓  – does this deferred assignment have any effect post 1.x? Now that we import `electron`, wouldn't that also import all its enumerable properties? (Like `systemPreferences`)